src/app/chip_data_model.cmake

```diff
@@ -119,6 +117,27 @@ function(chip_configure_data_model APP_TARGET)
         set(APP_GEN_FILES)
     endif()
 
+    # These are:
+    #   //src/app/icd/server:notfier
+    #   //src/app/icd/server:monitoring-table
+    #   //src/app/icd/server:configuration-data
+    #
+    # TODO: ideally we would avoid duplication and would link gn-built items. In this case
```

Comment:
I wonder if these are already not included in libCHIP.a. `libCHIP.a` already includes `src/app` as its public_deps, and `src/app` pulls ICD bits if `chip_enable_icd_server`. But I may be wrong as there are so many dependencies between the source sets that it's hard to statically analyse all the connections ;).

Reply:
I think we may want to start removing some bits here for sure. I just kept what was already there.

I would try to clean it up as another followup.